### PR TITLE
profiling: enable to capture dropped samples while buffering

### PR DESCRIPTION
**Pull request by mlim19**
Since the profiling utilizes the limited size of buffer to capture
sample data, dropping samples could happen while collecting data
if data is generated faster than flushing by consumer. Capturing
the dropped sample info is critical to understand how much the data
is reliable to use.

To capture the information, the new hypercall "PROFILING_GET_STATUS"
is introduced.

Tracked-On: #2474
Signed-off-by: Manisha Chinthapally <manisha.chinthapally@intel.com>
Signed-off-by: Min Lim <min.yeol.lim@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Comment by wenlingz** [comment]
@fyin1 please  help approve it, thanks.

File: hypervisor/debug/profiling.c [hunk 1]
```diff
@@ -1270,6 +1262,42 @@ int32_t profiling_get_pcpu_id(struct acrn_vm *vm, uint64_t addr)
...
+int32_t profiling_get_status_info(struct acrn_vm *vm, uint64_t gpa)
+{
+	uint16_t i;
+	struct profiling_status pstats[phys_cpu_num];
+
+	(void)memset((void *)&pstats, 0U, phys_cpu_num*sizeof(struct profiling_status));
+
+	dev_dbg(ACRN_DBG_PROFILING, "%s: entering", __func__);
+
+	if (copy_from_gpa(vm, &pstats, gpa,
```
Comment (fyin1) [inline hypervisor/debug/profiling.c]: Why do we need to copy these data from vm to HV? They are not used in HV actually.

Comment (mlim19) [reply hypervisor/debug/profiling.c]: Thanks for catching. You don't need it. I submit the revised patch.

**Comment by mlim19** [comment]
@fyin1, Can you please review the revised patch again?

**Comment by fyin1** [comment]
Sorry. I thought I approved the PR...

**Pull request merged**